pkg/log/logger.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 Outreach Corporation. All Rights Reserved.
+
+// Description: Implements a compatible API for the olog package and log package.
+
+package log
+
+import (
+	"context"
+	"io"
+	"log/slog"
+
+	"github.com/getoutreach/gobox/pkg/olog"
+	"github.com/sirupsen/logrus"
+)
+
+// Logger is the compatible log interface
+type Logger struct {
+	logrusLogger *logrus.Logger
+	ologLogger   *slog.Logger
+	useLogrus    bool
+}
+
+func New(useLogrus bool) *Logger {
+	if useLogrus {
+		l := logrus.New()
+		return &Logger{logrusLogger: l, useLogrus: true}
+	}
+	l := olog.New()
+	return &Logger{ologLogger: l, useLogrus: false}
+}
+
+func (log *Logger) Debug(ctx context.Context, message string, m Marshaler) {
+	if log.useLogrus {
+		Debug(ctx, message, m)
+	} else {
+		// add attr field to log message
+		log.ologLogger.DebugContext(ctx, message, "attr", m)
+	}
+}
+
+func (log *Logger) Info(ctx context.Context, message string, m Marshaler) {
+	if log.useLogrus {
+		Info(ctx, message, m)
+	} else {
+		log.ologLogger.InfoContext(ctx, message, "attr", m)
```

Comment:
question: how do marshaled attributes look in Datadog when using `pkg/log` vs `pkg/olog` when using this struct?